iphone/TitaniumKit/TitaniumKit/Sources/API/TiViewController.m

```diff
@@ -104,8 +104,11 @@ - (UIInterfaceOrientationMask)supportedInterfaceOrientations
      If we are in a navigation controller, let us match so it doesn't get freaked 
      out in when pushing/popping. We are going to force orientation anyways.
      */
-  if ([self navigationController] != nil) {
-    return [[self navigationController] supportedInterfaceOrientations];
+  /*
+   TIMOB-28282. Shouldn't UINavigationController.topViewController decide the supported orientation?
+   */
+  if ([self navigationController] != nil && [[self navigationController] topViewController] != self) {
```

Comment:
I do not know enough about this to give useful feedback. The code looks ok style-wise, but that's about all I can offer. 

I think for a navigation controller, you're supposed to ask it's delegate for the supported orientations:
- https://developer.apple.com/documentation/uikit/uinavigationcontroller?language=objc
- https://developer.apple.com/documentation/uikit/uinavigationcontrollerdelegate/1621884-navigationcontrollersupportedint?language=objc

But I don't see us implementing the expected method for the delegate protocol in any classes we declare conforming to the protocol.